File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        set<int>players, winners;
        map<int, int>loosers;
        
        for(vector<int>match : matches){
            winners.insert(match[0]);
            ++loosers[match[1]];
            
            players.insert(match[0]);
            players.insert(match[1]);
        }
        
        vector<int>win, lose;
        
        for(auto it : players){            
            if(loosers.find(it) == loosers.end())
                win.push_back(it);
        }
        
        for(auto it : loosers){
            if(it.second == 1)  
                lose.push_back(it.first);
        }
        
        return {win, lose};
    }
};

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.java
class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        Map<Integer, Integer> players = new TreeMap<>();

        for (int[] match : matches) {
            players.put(match[1], players.getOrDefault(match[1], 0) + 1);
        }

        for (int[] match : matches) {
            if (!players.containsKey(match[0])) {
                players.put(match[0], 0);
            }
        }

        List<Integer> win = new ArrayList<>();
        List<Integer> lose = new ArrayList<>();

        for (Map.Entry<Integer, Integer> entry : players.entrySet()) {
            if (entry.getValue() == 0) {
                win.add(entry.getKey());
            }

            if (entry.getValue() == 1) {
                lose.add(entry.getKey());
            }
        }

        List<List<Integer>> result = new ArrayList<>();
        result.add(win);
        result.add(lose);

        return result;
    }
}